extensions/amp-animation/0.1/web-animation-service.js

```diff
@@ -33,14 +33,15 @@ export class WebAnimationService {
   }
 
   /**
+   * @param {!Document|!ShadowRoot|!Element} root
    * @return {!Builder}
    */
-  createBuilder() {
+  createBuilder(root) {
     installWebAnimationsIfNecessary(this.ampdoc_.win);
 
     return new Builder(
       this.ampdoc_.win,
-      this.ampdoc_.getRootNode(),
+      root,
```

Comment:
If you go down this path, I'd definitely do `root || this.ampdoc_.getRootNode()`.